### PR TITLE
feat: add x-tagGroups in sidebar

### PR DESCRIPTION
**Pull request by marclave**

### CODE REVIEW

File: packages/api-reference/src/hooks/useSidebar.ts [hunk 1]
```diff
@@ -130,6 +131,30 @@ const items = computed(() => {
...
+    ? parsedSpec.value?.['x-tagGroups']?.map((tagGroup) => {
+        const children: SidebarEntry[] = []
+        tagGroup.tags.map((tagName: string) => {
+          const tag = operationEntries?.find(
+            (entry) => entry.title === tagName.toUpperCase(),
+          )
+          if (tag) {
+            children.push(tag)
+          }
+        })
```
Comment (amritk) [inline packages/api-reference/src/hooks/useSidebar.ts:146]: n^3 loops, I would re-write with one loop if possible

**Review by amritk — CHANGES_REQUESTED** [review]
Good for the most part, I would re-write that triple loop into one is possible. For something the size of the stripe spec it will destroy the cpu.

Also whats the difference between a `tag` and an `x-tagGroup`?

File: packages/api-reference/src/hooks/useSidebar.ts [hunk 2]
```diff
@@ -130,6 +131,30 @@ const items = computed(() => {
...
+        const sidebarTagGroup = {
+          id: tagGroup.name,
+          title: tagGroup.name.toUpperCase(),
+          children,
+          show: true,
+          isGroup: true,
+        }
+        return sidebarTagGroup
+      })
```
Comment (amritk) [inline packages/api-reference/src/hooks/useSidebar.ts:155]: This business

Comment (marclave) [reply packages/api-reference/src/hooks/useSidebar.ts:155]: @amritk imma be honest, i was cooking this at 1am and i had no idea what i was doing in es6 land 😭

Comment (amritk) [reply packages/api-reference/src/hooks/useSidebar.ts:155]: Alright once I got the spec, this code isn't the worst. BUT I will throw up a ticket for perf improvements on `useSidebar` as a whole as we can definitely combine some loops

**Review by amritk — APPROVED** [review]

**Pull request merged**